test/unit/mongo_logger.test.ts

```diff
@@ -1248,14 +1248,36 @@ describe('class MongoLogger', function () {
 
     context('when maxDocumentLength is non-zero', function () {
       context('when document has length greater than maxDocumentLength', function () {
-        it('truncates ejson string to length of maxDocumentLength + 3', function () {
-          expect(stringifyWithMaxLen(largeDoc, DEFAULT_MAX_DOCUMENT_LENGTH)).to.have.lengthOf(
-            DEFAULT_MAX_DOCUMENT_LENGTH + 3
-          );
+        context('when truncation does not occur mid-multibyte codepoint', function () {
+          it('truncates ejson string to length of maxDocumentLength + 3', function () {
+            expect(stringifyWithMaxLen(largeDoc, DEFAULT_MAX_DOCUMENT_LENGTH)).to.have.lengthOf(
+              DEFAULT_MAX_DOCUMENT_LENGTH + 3
+            );
+          });
         });
+
         it('ends with "..."', function () {
           expect(stringifyWithMaxLen(largeDoc, DEFAULT_MAX_DOCUMENT_LENGTH)).to.match(/^.*\.\.\.$/);
         });
+
+        context('when truncation occurs mid-multibyte codepoint', function () {
+          const multiByteCodePoint = '\ud83d\ude0d'; // heart eyes emoji
+          context('when maxDocumentLength = 1 but greater than 0', function () {
+            it('should return an empty string', function () {
+              expect(stringifyWithMaxLen(multiByteCodePoint, 1, { relaxed: true })).to.equal('');
+            });
+          });
+
+          context('when maxDocumentLength > 1', function () {
+            it('should round down maxDocLength to previous codepoint', function () {
+              const randomString = `random ${multiByteCodePoint}random random${multiByteCodePoint}${multiByteCodePoint}`;
+              const randomStringMinusACodePoint = `random ${multiByteCodePoint}random random${multiByteCodePoint}`;
```

Comment:
```suggestion
              const randomStringMinusACodePoint = `random ${multiByteCodePoint}random random${multiByteCodePoint}`;
              const randomString = `${randomStringMinusACodePoint}${multiByteCodePoint}`;
```

very optional, but this would make it a bit easier to see that `randomStringMinusACodePoint` really is a prefix of `randomString` by just looking at the test :)

Reply:
I see, that's a good idea! Just added it in